lib/libzfs/libzfs_dataset.c

```diff
@@ -2287,7 +2288,18 @@ get_numeric_property(zfs_handle_t *zhp, zfs_prop_t prop, zprop_source_t *src,
 		switch (zfs_prop_get_type(prop)) {
 		case PROP_TYPE_NUMBER:
 		case PROP_TYPE_INDEX:
-			*val = getprop_uint64(zhp, prop, source);
+			/*
+			 * We can directly read createtxg property from zfs
+			 * handle for filesystem and snapshot types.
+			 */
+			if ((prop == ZFS_PROP_CREATETXG) &&
+			    ((zhp->zfs_type == ZFS_TYPE_FILESYSTEM) ||
+			    (zhp->zfs_type == ZFS_TYPE_SNAPSHOT))) {
```

Comment:
What about `ZFS_TYPE_VOLUME`?

Reply:
You are right. Its handle is created within ```make_dataset_handle_common``` and we can make it use fast path as well. I will test and update.